git_toprepo.py

```diff
@@ -1509,13 +1519,14 @@ def get_commits_to_fetch() -> Set[CommitHash]:
                 commit_hash_str = commit_hash.decode("utf-8")
                 print(f"\trev-{commit_hash_str} = {url}")
             if len(missing_commits) > max_missing_commits_to_print:
-                print(f"Totally {len(missing_commits)} unexpected missing commits.")
-            print("ERROR: The referenced commits above could not be found")
+                print(f"{len(missing_commits)} unexpected missing commits.", file=sys.stderr)
+            print("Warning: The referenced commits above could not be found", file=sys.stderr)
             print(
                 "Either push the following commits or add them to the "
                 + "toprepo configuration."
             )
-            return None
+            if abort_on_missing:
+                sys.exit(1)
```

Comment:
Would serve us well to print a summary "Aborting due to previous warnings (--abort-on-unexpected-missing-commits)" before aborting.

Reply:
Please keep `return None` as the rest of the code base aborts through the call chain and not using `sys.exit`.
```suggestion
                return None
```

Reply:
Okay, if you prefer. Though if we treat it as a library just returning `None` is not a very clear signal that the caller should exit. And if the caller does not exit, then the error print that says "aborting" is misplaced. I have updated the return, but we should keep this in mind, that `None` is not customarily a clear signal to exit everything. Hmm.